### PR TITLE
Random failing test

### PR DESCRIPTION
**Pull request by karesti**
Try fix on ...

* computeIfPresent
* computeIfAbsent
* compute

... related tests in distributed mode

### CODE REVIEW

File: core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java [hunk 1]
```diff
@@ -306,27 +306,19 @@ public void testComputeFromNonOwner() throws InterruptedException {
       // compute function applied
       initAndTest();
       Object retval = getFirstNonOwner("k1").compute("k1", (k, v) -> "computed_" + k + "_" + v);
-      if (testRetVals) assertEquals("computed_k1_value", retval);
       asyncWait("k1", ComputeCommand.class);
-      assertOnAllCachesAndOwnership("k1", "computed_k1_value");
+      if (testRetVals) assertEquals("computed_k1_value", retval);
+      eventuallyAssertOnAllCachesAndOwneship("k1", "computed_k1_value");
```
Comment (rvansa) [inline core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java]: Why do you need the `eventually`? `asyncWait` should do the trick, shouldn't it?

Comment (karesti) [inline core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java]: well, it does not, apparently. I thought this "eventually" way makes thing more stable

Comment (rvansa) [inline core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java]: The flakiness might demonstrate a problem in the code that's not handled. Tests should not be fixed by loosening the requirements unless we know what's going on.

Comment (karesti) [inline core/src/test/java/org/infinispan/distribution/DistSyncFuncTest.java]: When I see a test that passes all the time for a while but in jenkins I get suddenly a => Expecting [k1] to equal [null] on cache [DistSyncFuncTest[DIST_SYNC, tx=false]-NodeB-50117] but was [computed_k1_value]. Owners are [Cache &apos;dist&apos;@DistSyncFuncTest[DIST_SYNC, tx=false]-NodeC-19559, Cache &apos;dist&apos;@DistSyncFuncTest[DIST_SYNC, tx=false]-NodeD-7512] 

telling me that a non owner has not deleted a value that was supposed to be deleted when looking directly in every cache, my guess is that the Command has not been executed yet as fast as it should be in every cache and the check is happening before that happens. The asyncWait of the command does nothing in the case of DistSyncFuncTest.

But you are right, the thing is that I'm not able to reproduce this problem. I'm going to open a ticket for this and check closer after I finish one thing I'm working on.

**Pull request closed without merging**